### PR TITLE
added tests

### LINKED ISSUE #594: testing: Write unit tests for the user input validator(isInputValid) in PDFUtils
The user input validator named `isInputValid` in PDFUtils is untested. It would be great if unit tests could be added with various test cases in order to test if the method works as expected.

Please write tests for the method with various user input cases.

### PR DESCRIPTION
**Pull request by Ni3verma**
# Description
added test for input validation in pdf split. Also changed the checkRangeValidity to **public static** for easy testing

Fixes #594 

## Type of change
Just put an x in the [] which are valid.
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes.
- [x] `./gradlew assembleDebug assembleRelease`
- [x] `./gradlew checkstyle`

# Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

### CODE REVIEW

**Comment by Ni3verma** [comment]
@srv-twry please check, am I doing it correct? I'll add more tests later.
and I had to make **checkRangeValidity** to public static so that I can call it in test file.

**Comment by sidhuparas** [comment]
The tests look fine to me. I think the name of the test file should be like `PDFUtilsTest.java` instead of `CheckRangeTest.java` because we can't have a whole file for a single function's test. Along with this, the change you mentioned of changing `checkRangeValidity` to `public static` can be mentioned in description so that it can be traced back easily if required. Although it's a very small change but worth mentioning. Waiting for inputs from @srv-twry and @vyankatesh24.

File: app/src/main/java/swati4star/createpdf/util/PDFUtils.java [hunk 1]
```diff
@@ -620,7 +620,7 @@ private boolean isInputValid(String path, String[] ranges) {
      * ERROR_PAGE_NUMBER if range greater than max number of pages
      * ERROR_PAGE_RANGE if range is invalid like 9-4
      */
-    private int checkRangeValidity(int numOfPages, String[] ranges) {
+    public static int checkRangeValidity(int numOfPages, String[] ranges) {
```
Comment (Swati4star) [inline app/src/main/java/swati4star/createpdf/util/PDFUtils.java]: Do we need to make it static?

File: app/src/test/java/util/CheckRangeTest.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    @Test
+    public void checkRange_CorrectRanges_ReturnsZero() {
+        String[] ranges = {"1", "2", "3-8"};
+        assertThat(PDFUtils.checkRangeValidity(NUM_OF_PAGES, ranges), is(0));
+    }
+
+    @Test
+    public void checkRange_IncorrectPageNumber_ReturnsOne() {
+        String[] ranges = {"1", "2", "24"}; //24 > max number of pages
+        assertThat(PDFUtils.checkRangeValidity(NUM_OF_PAGES, ranges), is(1));
```
Comment (Swati4star) [inline app/src/test/java/util/CheckRangeTest.java]: instead of using the numbers, use enum.
and for valid range also, instead of using '0', use a different enum

File: app/src/test/java/util/CheckRangeTest.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    }
+
+    @Test
+    public void checkRange_IncorrectRange_ReturnsTwo() {
+        String[] ranges = {"1", "2", "3-1"};  //invalid range 3-1
```
Comment (Swati4star) [inline app/src/test/java/util/CheckRangeTest.java]: add one more test with negative numbers.

File: app/src/main/java/swati4star/createpdf/util/PDFUtils.java (hunk 1 above)
Comment (srv-twry) [reply app/src/main/java/swati4star/createpdf/util/PDFUtils.java]: I think making it static would make things easier for him. In case of it being non-static, he'll have to create an instance of PDFUtils class which expects a context in the constructor. As a result he'll have to mock an activity as well. So making it static, solves that issue.

File: app/src/test/java/util/CheckRangeTest.java [hunk 4]
```diff
@@ -0,0 +1,30 @@
+package util;
+
+import org.junit.Test;
+
+import swati4star.createpdf.util.PDFUtils;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+public class CheckRangeTest {
```
Comment (srv-twry) [inline app/src/test/java/util/CheckRangeTest.java]: Change the name of the file and the class to `PDFUtilsTest.java`

File: app/src/test/java/util/CheckRangeTest.java (hunk 2 above)
Comment (Ni3verma) [reply app/src/test/java/util/CheckRangeTest.java]: okay

File: app/src/test/java/util/CheckRangeTest.java (hunk 4 above)
Comment (Ni3verma) [reply app/src/test/java/util/CheckRangeTest.java]: okay

**Comment by Ni3verma** [comment]
@Swati4star @srv-twry @vyankatesh24 @sidhuparas 
negative number crashes the app.
should I make a new issue and solve it first?

**Comment by Ni3verma** [comment]
negative numbers, double commas, multiple dash...and there are so many other issues.
I think I should add more validation checks.
should I solve it in a different issue or here? @Swati4star

**Comment by Swati4star** [comment]
@Ni3verma, you should have taken care of those cases also before.
Now, first you can rectify the function to handle these cases (make a PR for it), and then continue with the current PR

**Comment by Ni3verma** [comment]
made some changes in string resource files as now o and pages greater than max number of pages come under invalid page number.
added if condition to check any page input is **zero**

**Comment by Ni3verma** [comment]
Like me, I guees everyone forgot about testing with **zero**.. and tested the app with negative numbers,dashes and commas...
So I had to make these extra changes

File: app/src/test/java/util/PDFUtilsTest.java [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+
+public class PDFUtilsTest {
+    private static final int NUM_OF_PAGES = 23;
+    private static final int NO_ERROR = 0;
+    private static final int ERROR_PAGE_NUMBER = 1;
+    private static final int ERROR_RANGE = 2;
+    private static final int ERROR_INVALID_INPUT = 3;
+
+    @Test
+    public void checkRange_CorrectRanges_Returns_NO_ERROR() {
```
Comment (srv-twry) [inline app/src/test/java/util/PDFUtilsTest.java]: Don't use snake case please. It's Java and not Python.

Comment (Ni3verma) [reply app/src/test/java/util/PDFUtilsTest.java]: But in java constants are named like this.. that's why I did so.
AndI named function like this because in fileUtilsTest.java file, it is done like this.

Comment (srv-twry) [reply app/src/test/java/util/PDFUtilsTest.java]: > But in java constants are named like this.

Yes, but they are all caps and this is not a constant.

> AndI named function like this because in fileUtilsTest.java file, it is done like this.

Then that is wrong. You can create an issue and fix it.

Comment (Ni3verma) [reply app/src/test/java/util/PDFUtilsTest.java]: Just to confirm.
I only need to make change in test **name** and make it camelCase?

Comment (Swati4star) [reply app/src/test/java/util/PDFUtilsTest.java]: yes. just make these camel case

Comment (Ni3verma) [reply app/src/test/java/util/PDFUtilsTest.java]: done

Comment (Ni3verma) [reply app/src/test/java/util/PDFUtilsTest.java]: @Swati4star should I make a new issue for wrong naming in fileUtilsTest.java or you will do it?

Comment (Swati4star) [reply app/src/test/java/util/PDFUtilsTest.java]: leave that. I'll change those later
